rasa/core/policies/rule_policy.py

```diff
@@ -519,6 +519,37 @@ def _handling_loop_sources(domain: Domain) -> Set[Text]:
             )
         return loop_sources
 
+    def _should_delete(
+        self,
+        prediction_source: Text,
+        tracker: TrackerWithCachedStates,
+        gold_action_name: Text,
+    ) -> bool:
+        """Checks whether this contradiction is due to action, intent pair.
+
+        Args:
+            prediction_source: the states that result in the prediction
+            tracker: the tracker that raises the contradiction
+
+        Returns:
+            true if the contradiction is a result of an action, intent pair in the rule.
+        """
+        if (
+            tracker.is_rule_tracker  # only apply to contradicting story, not rule
+            or prediction_source.count(PREVIOUS_ACTION)
+            > 1  # only apply for prediction after unpredictable action
+            or gold_action_name
+            != ACTION_LISTEN_NAME  # only apply for prediction of action_listen
+        ):
```

Comment:
I think this way it is a bit prettier
```suggestion
        if (
            # only apply to contradicting story, not rule
            tracker.is_rule_tracker
            # only apply for the first prediction in a rule
            or prediction_source.count(PREVIOUS_ACTION) > 1
            # only apply for prediction of action_listen
            or gold_action_name != ACTION_LISTEN_NAME
        ):
```